File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int n = nums.size();
        vector<int> pre(n);
        pre[0] = nums[0];
        for(int i = 1; i<n; i++) pre[i] = pre[i - 1] + nums[i];
        int i = 0, j = 0, ans = INT_MAX;
        while(i <= j && j < n) {
            if(pre[j] - pre[i] + nums[i] >= target) {
                ans = min(ans, j - i + 1);
                i++;
            }
            else j++;
        }
        return ans == INT_MAX ? 0 : ans;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        pre = [0] * n
        pre[0] = nums[0]
        for i in range(1, n):
            pre[i] = pre[i - 1] + nums[i]
        l = 0
        r = 0
        ans = 10 ** 20
        while l <= r and r < n:
            if pre[r] - pre[l] + nums[l] >= target:
                ans = min(ans, r - l  + 1)
                l += 1
            else:
                r += 1
        if ans == 10 ** 20:
            return 0
        return ans